Sources/WendyAgent/Services/WendyAgentService.swift

```diff
@@ -547,3 +551,70 @@ struct WendyAgentService: Wendy_Agent_Services_V1_WendyAgentService.ServiceProto
         }
     }
 }
+
+/// Validates that a file is a valid ELF executable binary before applying an update
+/// This prevents bricking the agent with corrupted or invalid binaries
+private func validateBinaryFormat(at path: FilePath, filesystem: FileSystem) async throws {
+    // Read first 64 bytes for ELF header validation
+    let headerBytes = try await filesystem.withFileHandle(forReadingAt: path) { handle in
+        var buffer = try await handle.read(fromAbsoluteOffset: 0, maximumSizeInBytes: .bytes(64))
+        return buffer.readBytes(length: buffer.readableBytes) ?? []
```

Comment:
```suggestion
        let buffer = try await handle.read(fromAbsoluteOffset: 0, maximumSizeInBytes: .bytes(64))
        return [UInt8](buffer: buffer)
```

I believe this should work. String and Data among other types _definitely_ have initializers to take the data out of a bytebuffer _efficiently_. I believe Array does too